test/TestShaderResources.cpp

```diff
@@ -0,0 +1,45 @@
+#include "gtest/gtest.h"
+
+#include "kompute/Kompute.hpp"
+
+const std::string shaderString = (R"(
+    #version 450
+
+    layout (local_size_x = 1) in;
+
+    // The input tensors bind index is relative to index in parameter passed
+    layout(set = 0, binding = 0) buffer bina { float tina[]; };
+    layout(set = 0, binding = 1) buffer binb { float tinb[]; };
+    layout(set = 0, binding = 2) buffer bout { float tout[]; };
+
+    void main() {
+        uint index = gl_GlobalInvocationID.x;
+        
+        int i = 1;
+        
+        while(i < 200) {
+            tout[index] += tina[index] * tinb[index];
+            i++;
+        }
+    }
+)");
+
+void compileShaderWithGivenResources(const std::string shaderString, const TBuiltInResource resources) {
+    kp::Shader::compile_source(shaderString,  std::string("main"), std::vector<std::pair<std::string,std::string>>({}), resources);
+}
+
+
+
+TEST(TestShaderResources, TestNoMaxLight)
+{
+    const TBuiltInResource noMaxLightResources = { /* .MaxLights = */ 0, 6, 32, 32, 64, 4096, 64, 32, 80, 32, 4096, 32, 128, 8, 16, 16, 15, -8, 7, 8, 65535, 65535, 65535, 1024, 1024, 64, 1024, 16, 8, 8, 1, 60, 64, 64, 128, 128, 8, 8, 8, 0, 0, 0, 0, 0, 8, 8, 16, 256, 1024, 1024, 64, 128, 128, 16, 1024, 4096, 128, 128, 16, 1024, 120, 32, 64, 16, 0, 0, 0, 0, 8, 8, 1, 0, 0, 0, 0, 1, 1, 16384, 4, 64, 8, 8, 4, 256, 512, 32, 1, 1, 32, 1, 1, 4, 1, {true, true, true, true, true, true, true, true, true}};
```

Comment:
Question, is it possible to use the defaultResource as base and only modify the specific value? Could make it easier to maintain - if not we can go ahead with this approach

Reply:
Yes, TBuiltInResource is a struct. But we have to change defaultResource to a variable instead of a const, and it should be a global variable, so that means we no longer have to put it inside a class.
I actually like this idea, because `const TBuiltInResource noMaxLightResources = { /* .MaxLights = */ 0, 6, 32,  ...` will turn into: `defaultResource.maxLights=0`, which is more intuitive and makes more sense.

Reply:
EDIT: another option is to keep defaultResource as a const, but then create a copy of it and then modify the values.

Reply:
I agree with you that it would be quite nice if it can be edited as struct - the latter would make sense as when you assign the struct it would create a copy by value which means you would be able to modify the object

Reply:
This means that inside of the test, the variables don't have to be static const - they can just be normal values, so you can modify them accordingly.

Reply:
Sounds good

Reply:
It's much better now